x-pack/plugins/alerting/server/routes/create_rule.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { validateDurationSchema, ILicenseState, AlertTypeDisabledError } from '../lib';
+import { CreateOptions } from '../alerts_client';
+import {
+  RewriteRequestCase,
+  RewriteResponseCase,
+  handleDisabledApiKeysError,
+  verifyAccessAndContext,
+} from './lib';
+import {
+  SanitizedAlert,
+  validateNotifyWhenType,
+  AlertTypeParams,
+  AlertingRequestHandlerContext,
+  BASE_ALERTING_API_PATH,
+  AlertNotifyWhenType,
+} from '../types';
+
+export const bodySchema = schema.object({
+  name: schema.string(),
+  rule_type_id: schema.string(),
+  enabled: schema.boolean({ defaultValue: true }),
+  consumer: schema.string(),
+  tags: schema.arrayOf(schema.string(), { defaultValue: [] }),
+  throttle: schema.nullable(schema.string({ validate: validateDurationSchema })),
+  params: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
+  schedule: schema.object({
+    interval: schema.string({ validate: validateDurationSchema }),
+  }),
+  actions: schema.arrayOf(
+    schema.object({
+      group: schema.string(),
+      id: schema.string(),
+      params: schema.recordOf(schema.string(), schema.any(), { defaultValue: {} }),
+    }),
+    { defaultValue: [] }
+  ),
+  notify_when: schema.string({ validate: validateNotifyWhenType }),
+});
+
+const rewriteBodyReq: RewriteRequestCase<CreateOptions<AlertTypeParams>['data']> = ({
+  rule_type_id: alertTypeId,
+  notify_when: notifyWhen,
+  ...rest
+}) => ({
+  ...rest,
+  alertTypeId,
+  notifyWhen,
+});
+const rewriteBodyRes: RewriteResponseCase<SanitizedAlert<AlertTypeParams>> = ({
+  actions,
+  alertTypeId,
+  scheduledTaskId,
+  createdBy,
+  updatedBy,
+  createdAt,
+  updatedAt,
+  apiKeyOwner,
+  notifyWhen,
+  muteAll,
+  mutedInstanceIds,
+  executionStatus: { lastExecutionDate, ...executionStatus },
+  ...rest
+}) => ({
+  ...rest,
+  rule_type_id: alertTypeId,
+  scheduled_task_id: scheduledTaskId,
+  created_by: createdBy,
+  updated_by: updatedBy,
+  created_at: createdAt,
+  updated_at: updatedAt,
+  api_key_owner: apiKeyOwner,
+  notify_when: notifyWhen,
+  mute_all: muteAll,
+  muted_instance_ids: mutedInstanceIds,
```

Comment:
Should this be `muted_alert_ids` to follow the new terminology?

Reply:
Good catch! Fixed in https://github.com/elastic/kibana/pull/93980/commits/075315503ebd0def045a4cd8004d6a15e2de5084.